airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -1808,6 +1808,8 @@ def __init__(
         gcp_conn_id: str = "google_cloud_default",
         impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
         asynchronous: bool = False,
+        hook: Optional[DataprocHook] = None,
```

Comment:
Why do you want to create hook object within the `__init__` method?

Reply:
@michalslowikowski00 > Why do you want to create hook object within the `__init__` method?

 I wanted `DataprocSubmitJobOperator` to be more customizable, and I need `hook` and `job_id`  as properties to be able to cancel job in `on_kill` method.

If that's better I can disallow to initialise these properties in `__init__`.

Reply:
The general idea is that operators should initialize the hook in execute methods. What users should pass in operators constructor is parameters for hook initialization (in this case `gcp_conn_id` and `impersonation_chain`).

This operator can use only `DataprocHook` using any other hook may result in strange errors. Recommended way to to create "custom behavior" in Airflow is creating new operators.